WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -135,6 +151,59 @@ class NotificationsViewController: UITableViewController {
         tableViewHandler.updateRowAnimation = .none
     }
 
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if splitViewControllerIsHorizontallyCompact {
+            tableView.deselectSelectedRowWithAnimation(true)
+        } else {
+            selectRowForNotification(selectedNotification, animated: true, scrollPosition: .middle)
+        }
+    }
+
+    // MARK: - State Restoration
+
+    static func viewController(withRestorationIdentifierPath identifierComponents: [Any], coder: NSCoder) -> UIViewController? {
+        return WPTabBarController.sharedInstance().notificationsViewController
+    }
+
+    override func encodeRestorableState(with coder: NSCoder) {
+        if let uriRepresentation = selectedNotification?.objectID.uriRepresentation() {
+            coder.encode(uriRepresentation, forKey: type(of: self).selectedNotificationRestorationIdentifier)
+        }
+
+        coder.encode(filtersSegmentedControl.selectedSegmentIndex, forKey: type(of: self).selectedSegmentIndexRestorationIdentifier)
+
+        super.encodeRestorableState(with: coder)
+    }
+
+    override func decodeRestorableState(with coder: NSCoder) {
+        decodeSelectedSegmentIndexWithCoder(coder: coder)
+        decodeSelectedNotificationWithCoder(coder: coder)
+
+        reloadResultsController()
+
+        super.decodeRestorableState(with: coder)
+    }
+
+    fileprivate func decodeSelectedNotificationWithCoder(coder: NSCoder) {
```

Comment:
Can probably make this a bit more Swift-3-y. `decodeSelectedNotification(coder:`.